From 5825541216d49668aa7d19fdffc4f5519e2f5ff0 Mon Sep 17 00:00:00 2001
From: Phil Sutter <psutter@redhat.com>
Date: Wed, 8 May 2024 22:39:40 +0200
Subject: [PATCH] set: Validate NFTNL_SET_ID, too

JIRA: https://issues.redhat.com/browse/RHEL-28515
Upstream Status: libnftnl commit a9b4d07dfab235324d2efbaa242fcc5ed5efe4c1

commit a9b4d07dfab235324d2efbaa242fcc5ed5efe4c1
Author: Phil Sutter <phil@nwl.cc>
Date:   Thu Mar 14 17:29:51 2024 +0100

    set: Validate NFTNL_SET_ID, too

    Fixes: 26298a9ffc2e2 ("set: add set ID support")
    Signed-off-by: Phil Sutter <phil@nwl.cc>

Signed-off-by: Phil Sutter <psutter@redhat.com>
---
 src/set.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/set.c b/src/set.c
index b51ff9e..a732bc0 100644
--- a/src/set.c
+++ b/src/set.c
@@ -128,6 +128,7 @@ static uint32_t nftnl_set_validate[NFTNL_SET_MAX + 1] = {
 	[NFTNL_SET_DATA_LEN]		= sizeof(uint32_t),
 	[NFTNL_SET_OBJ_TYPE]		= sizeof(uint32_t),
 	[NFTNL_SET_FAMILY]		= sizeof(uint32_t),
+	[NFTNL_SET_ID]			= sizeof(uint32_t),
 	[NFTNL_SET_POLICY]		= sizeof(uint32_t),
 	[NFTNL_SET_DESC_SIZE]	= sizeof(uint32_t),
 	[NFTNL_SET_TIMEOUT]		= sizeof(uint64_t),
